src/token/TokenRejectTransaction.js

```diff
@@ -0,0 +1,295 @@
+            this._tokenIds = props.tokenIds;
+        }
+
+        if (props.nftIds != null) {
+            this._nftIds = props.nftIds;
+        }
+    }
+
+    /**
+     * @internal
+     * @param {HashgraphProto.proto.ITransaction[]} transactions
+     * @param {HashgraphProto.proto.ISignedTransaction[]} signedTransactions
+     * @param {TransactionId[]} transactionIds
+     * @param {AccountId[]} nodeIds
+     * @param {HashgraphProto.proto.ITransactionBody[]} bodies
+     * @returns {TokenRejectTransaction}
+     */
+    static _fromProtobuf(
+        transactions,
+        signedTransactions,
+        transactionIds,
+        nodeIds,
+        bodies,
+    ) {
+        const body = bodies[0];
+        const rejectToken =
+            /** @type {HashgraphProto.proto.ITokenRejectTransactionBody} */ (
+                body.tokenReject
+            );
+
+        const tokenIds = rejectToken.rejections?.map((rejection) =>
+            TokenReference._fromProtobuf(rejection),
+        );
+        const ftIds = tokenIds
+            ?.filter((token) => token.tokenIdentifier === "fungibleToken")
+            .map(({ fungibleToken }) => {
+                if (fungibleToken == null) {
+                    throw new Error("Fungible Token cannot be null");
+                }
+                return fungibleToken;
+            });
+
+        const nftIds = tokenIds
+            ?.filter((token) => token.tokenIdentifier === "nft")
+            .map(({ nft }) => {
+                if (nft == null) {
+                    throw new Error("Nft cannot be null");
+                }
+                return nft;
+            });
+
+        return Transaction._fromProtobufTransactions(
+            new TokenRejectTransaction({
+                owner:
+                    rejectToken.owner != null
+                        ? AccountId._fromProtobuf(rejectToken.owner)
+                        : undefined,
+
+                tokenIds: ftIds,
+                nftIds: nftIds,
+            }),
+            transactions,
+            signedTransactions,
+            transactionIds,
+            nodeIds,
+            bodies,
+        );
+    }
+
+    /**
+     * @returns {TokenId[] | string[]}
+     * @readonly
+     */
+    get tokenIds() {
+        return this._tokenIds;
+    }
+
+    /**
+     * @param {TokenId[] | string[]} tokenIds
+     * @returns {this}
+     */
+    setTokenIds(tokenIds) {
+        this._requireNotFrozen();
+        this._tokenIds = tokenIds.map((tokenId) =>
+            tokenId instanceof TokenId ? tokenId : TokenId.fromString(tokenId),
+        );
+        return this;
+    }
+
+    /**
+     * @param {TokenId | string} tokenId
+     * @returns {this}
+     */
+    addTokenId(tokenId) {
+        this._requireNotFrozen();
+        this._tokenIds?.push(
+            tokenId instanceof TokenId ? tokenId : TokenId.fromString(tokenId),
+        );
+        return this;
+    }
+
+    /**
+     * @returns {NftId[] | string[]}
```

Comment:
The result will be `NftId[]` always as it declared in the constructor.
```suggestion
     * @returns {NftId[]}
```